docs/source/tutorial/test_a_project.md

```diff
@@ -0,0 +1,335 @@
+
+Writing tests for each node ensures each node will behave as expected when run individually. However, we must consider how nodes in a pipeline interact with each other - this is called integration testing. Integration testing combines individual units as a group and checks whether they communicate correctly, share data properly, and work as expected when combined. Let us look at this in practice.
+
+Consider the data science pipeline as a whole:
+
+<details>
+<summary><b>Click to expand</b></summary>
+
+```python
+from kedro.pipeline import Pipeline, node, pipeline
+from .nodes import evaluate_model, split_data, train_model
+
+
+def create_pipeline(**kwargs) -> Pipeline:
+    return pipeline(
+        [
+            node(
+                func=split_data,
+                inputs=["model_input_table", "params:model_options"],
+                outputs=["X_train", "X_test", "y_train", "y_test"],
+                name="split_data_node",
+            ),
+            node(
+                func=train_model,
+                inputs=["X_train", "y_train"],
+                outputs="regressor",
+                name="train_model_node",
+            ),
+            node(
+                func=evaluate_model,
+                inputs=["regressor", "X_test", "y_test"],
+                outputs=None,
+                name="evaluate_model_node",
+            ),
+        ]
+    )
+```
+</details>
+
+The pipeline takes a DataFrame and dictionary of parameters as input, splits the data in accordance to the parameters, and uses it to train and evaluate a regression model. With an integration test, we can validate that this sequence of nodes runs as expected. As we did with our unit tests, we break this down into several steps:
+
+1. Arrange: Prepare the runner and its inputs `pipeline` and `catalog`.
+2. Act: Run the pipeline.
+3. Assert: Ensure the outputs are as expected.
+
+When we put this together, we get the following test:
+
+<details>
+<summary><b>Click to expand</b></summary>
+
+```python
+# NOTE: This example test is yet to be refactored.
+# A complete version is available under the testing best practices section.
+
+import pandas as pd
+from kedro.io import DataCatalog, MemoryDataset
+from kedro.runner import SequentialRunner
+from spaceflights.pipelines.data_science import create_pipeline
+
+    def test_data_science_pipeline():
+        # Arrange pipeline
+        pipeline = create_pipeline()
+
+        # Arrange data catalog
+        catalog = DataCatalog()
+
+        dummy_data = pd.DataFrame(
+        {"engines": [1, 2, 3],
+         "crew": [4, 5, 6],
+         "passenger_capacity": [5, 6, 7],
+         "price": [120, 290, 30]})
+        catalog.add("model_input_table", MemoryDataset(dummy_data))
+
+        dummy_parameters = {"model_options":
+                     {"test_size": 0.2,
+                      "random_state": 3,
+                      "features": ["engines", "passenger_capacity", "crew"]}}
```

Comment:
If we use explicit (`params:x`) this is much more readable

Reply:
I believe all the calls in the tests use the explicit `params:model_options`, is there somewhere else this could be used?